src/smif/data_layer/data_array.py

```diff
@@ -209,16 +209,37 @@ def validate_as_full(self):
         """Check that the data array contains no NaN values
         """
         if np.issubdtype(self.data.dtype, np.number):
-            if np.any(np.isnan(self.data)):
-                raise SmifDataMismatchError
+            mask = np.zeros(self.data.shape, dtype=np.bool)
+            np.isnan(self.data, out=mask)
+
+            if np.any(mask):
+                missing_data = self._show_na()
```

Comment:
Can this message/log section be pulled out of the conditional? Maybe something like:
```suggestion
        if np.issubdtype(self.data.dtype, np.number):
            mask = np.zeros(self.data.shape, dtype=np.bool)
            np.isnan(self.data, out=mask)
        else:
            # create vectorised test for nan to use against np.array with dtype=object
            def _is_nan(x):
                return x is np.nan
            _is_nan = np.frompyfunc(_is_nan, 1, 1)
            mask = _is_nan(self.data).astype(bool)
            
        if np.any(mask):
            self.logger.debug("Missing data:\n\n    %s", self._show_na())
            msg = "There are missing data points in '{}'"
            raise SmifDataMismatchError(msg.format(self.name))
```

Reply:
PS not sure if that suggested change is quite right, just trying out the feature.